platalea/dataset.py

```diff
@@ -52,17 +53,19 @@ def __init__(self, root, feature_fname, split='train', language='en',
             raise ValueError('Language {} not supported.'.format(language))
         self.root = root
         self.split = split
-        fmeta = open(root + 'dataset_multilingual.json')
-        self.metadata = json.load(fmeta)['images']
+        root_path = pathlib.Path(root)
+        with open(root_path / platalea.config.args.meta) as fmeta:
+            self.metadata = json.load(fmeta)['images']
         if downsampling_factor is not None:
             num_examples = len(self.metadata) // downsampling_factor
             self.metadata = random.sample(self.metadata, num_examples)
         # mapping from image id to list of caption id
         self.image_captions = {}
-        for line in open(os.path.join(root + 'flickr_audio/wav2capt.txt')):
-            audio_id, image_id, text_id = line.split()
-            text_id = int(text_id[1:])
-            self.image_captions[image_id] = self.image_captions.get(image_id, []) + [(text_id, audio_id)]
+        with open(root_path / 'wav2capt.txt') as fwav2capt:
```

Comment:
wav2capt.txt is by default under flickr_audio folder
```suggestion
        with open(root_path / 'flickr_audio' / 'wav2capt.txt') as fwav2capt:
```

Reply:
Ah, yes, actually it was both in the flickr8k folder and in the flickr_audio sub folder, they were duplicates. I'm fine with either choice, under audio makes sense. Should then change that in flickr1d as well.